src/core/container.rs

```diff
@@ -128,13 +133,17 @@ impl<'d, I> Container<'d, I> {
     /// This method panics if the given port is not mapped.
     /// Testcontainers is designed to be used in tests only. If a certain port is not mapped, the container
     /// is unlikely to be useful.
-    pub fn get_host_port(&self, internal_port: u16) -> u16 {
+    pub fn get_host_port<T>(&self, internal_port: T) -> T
```

Comment:
How does one go about calling this API? From what I can see, everything that implements `MapToHostPort` is actually private and can't be used by users outside of the library.

To make sure this doesn't happen, can you add an example in `examples/` (dir doesn't exist yet). Rust examples are compiled as part of the tests so that will be a good way of ensuring people can actually use the API :)